pkg/websocket/connection.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package websocket
+
+import (
+	"bytes"
+	"encoding/gob"
+	"time"
+
+	"github.com/gorilla/websocket"
+)
+
+// Connection represents a websocket connection.
+type Connection struct {
+	target     string
+	connection *websocket.Conn
+}
+
+func connect(target string) *websocket.Conn {
+	for {
+		// TODO: use exponential backoff here
```

Comment:
Go ahead and reference #1843 here.  E.g. `TODO(#1843): `

Reply:
I'll leave this as is, as I expect #1925 to go in before this, exponential backoff is handled there.